### ISSUE TITLE
[BUG] Error: "Please select no more than one LCD controller option." Wanhao D6

### ISSUE THREAD

**Issue by Deneteus** [MarlinFirmware/Marlin#16087]
**### Bug Description**

PlatformIO Build fails due to multiple errors when compiling using the Wanhao D6 Example configuration files included with Marlin 2.0 

**### My Configurations**

I used the stock example configuration files for the Wanhao D6 and they failed to compile.
\Marlin\config\examples\Wanhao\Duplicator 6\

### Steps to Reproduce

1. Copy the Wanhao D6 example configuration files to the Marlin project folder.
2. Open the Marlin project in Atom and confirm the configuration.h file is the correct file.
3. Under Platform IO Home choose Inspect and make sure the Project Environment is megatmega2560
4. Choose PlatformIO on the tool bar and select Build.
5.  Watch as the Sanity Check Error occurs and the build fails.

**Expected behavior:** Example Configuration should compile with no errors.

**Actual behavior:** Multiple sanity check errors occur related to multiple LCD controller options. 

#### Additional Information

Just to make sure my config had not been corrupted from the previous version I wiped out my Marlin folder and redownloaded it.

The software updates and updates to libraries had already been done before this.
The printer is currently running an older version of Marlin 2.0 from earlier this year.

![image](https://user-images.githubusercontent.com/6380390/70075972-71c89c00-15c3-11ea-96c2-51eb1fe40307.png)

![image](https://user-images.githubusercontent.com/6380390/70075448-7476c180-15c2-11ea-9dcf-1f53ebc74221.png)

I am assuming its this line that's causing the issue.

![image](https://user-images.githubusercontent.com/6380390/70076914-65454300-15c5-11ea-99d9-409b20ba8281.png)

**Comment by thisiskeithb** [comment]
Both `ULTIPANEL` and `U8GLIB_SSD1306` are defined in that config. Comment out `ULTIPANEL` and see if your LCD still works after updating.

**Comment by Deneteus** [comment]
> Both `ULTIPANEL` and `U8GLIB_SSD1306` are defined in that config. Comment out `ULTIPANEL` and see if your LCD still works after updating.

Disabling Ultipanel causes this: 

![image](https://user-images.githubusercontent.com/6380390/70159924-4b683680-167f-11ea-8921-e99d1a20f373.png)

I disabled advanced_pause in the advanced config file and I was able to complete the build.  

There was one error with the LiquidCrystal_SR1W.cpp and one with the DogM.cpp. 

I was able to fix the Liquid Crystal and DogM error by changing the slash / in the line to another character. An entire schematic in that file is a bit much. LOL

![image](https://user-images.githubusercontent.com/6380390/70161884-67210c00-1682-11ea-8450-afb81d1cba16.png)

![image](https://user-images.githubusercontent.com/6380390/70160724-93d42400-1680-11ea-8a42-905e13ca7998.png)

![image](https://user-images.githubusercontent.com/6380390/70160635-6f784780-1680-11ea-9132-0c371356a681.png)

**Comment by boelle** [comment]
Configurations, please
Please ZIP up your `Configuration.h` and `Configuration_adv.h` files (as
requested in the Issue template) and drop them into your next reply.
We'll check them over and see if anything is amiss.

**Comment by thisiskeithb** [comment]
> Configurations, please
> Please ZIP up your `Configuration.h` and `Configuration_adv.h` files (as
> requested in the Issue template) and drop them into your next reply.
> We'll check them over and see if anything is amiss.

They are using the stock D6 configs:

> I used the stock example configuration files for the Wanhao D6 and they failed to compile.
\Marlin\config\examples\Wanhao\Duplicator 6

**Comment by boelle** [comment]
> They are using the stock D6 configs:

did not read that, just saw a missing zip

**Comment by Deneteus** [comment]
I figured there would be no point in attaching the stock files.
[Uploading Wanhao D6 Marlin 2.0 Stock Example Configs.zip…]()

**Comment by boelle** [comment]
as said i did not read the post just looked if something was attached

but yeah standard config not working is kind of embarresing,

**Comment by Deneteus** [comment]
I tested using the firmware as compiled after those two changes and it failed.

**Comment by shitcreek** [comment]
The problem is line 1991: `// #define U8GLIB_SSD1306`
I've made the correction. Thanks.

**Comment by thisiskeithb** [comment]
Does this work with only `ULTIPANEL` defined? I thought `U8GLIB_SSD1306` was required since this printer has an OLED screen. 

There are extra options underneath `U8GLIB_SSD1306` in the D6 config compared to the stock config which makes me think it's still required:
https://github.com/MarlinFirmware/Marlin/blob/f64649933f3a2195df7a2bd3ec95ab6ed3885975/config/examples/Wanhao/Duplicator%206/Configuration.h#L1991-L1994

Stock config:
https://github.com/MarlinFirmware/Marlin/blob/f64649933f3a2195df7a2bd3ec95ab6ed3885975/config/default/Configuration.h#L1978-L1982

**Comment by shitcreek** [comment]
It compiled without issue. @Deneteus mind testing this out to confirm it works with `U8GLIB_SSD1306` undefined?

**Comment by thisiskeithb** [comment]
Digging back into the origins of this config, it came with both those LCD types enabled from the start. I wonder if something changed in `SanityCheck.h` for this to start throwing errors or if this config ever worked in the first place.

**Comment by Deneteus** [comment]
@shitcreek I had not tested it with the SSD1306 OLED LCD undefined.
@thisiskeithb  The motherboard for the D6 is an Ultimaker 2 clone. It is based on the ATMega2560 microcontroller. 

![image](https://user-images.githubusercontent.com/6380390/70195086-3cf33c80-16ca-11ea-9303-4165e6ea1a24.png)

**Comment by thisiskeithb** [comment]
If that doesn't work, it seems this [Marlin port]( https://github.com/dot-bob/Marlin-Duplicator-6) is popular and they have a different LCD defined:

```cpp
//
// TinyBoy2 128x64 OLED / Encoder Panel
//
#define OLED_PANEL_TINYBOY2
#define LCD_RESET_PIN 5
```

**Comment by Deneteus** [comment]
> If that doesn't work, it seems this [Marlin port](https://github.com/dot-bob/Marlin-Duplicator-6) is popular and they have a different LCD defined:
> 
> ```c++
> //
> // TinyBoy2 128x64 OLED / Encoder Panel
> //
> #define OLED_PANEL_TINYBOY2
> #define LCD_RESET_PIN 5
> ```

I'll test that in a minute!

**Comment by thisiskeithb** [comment]
This config builds with a single LCD defined now (as expected), so we just need confirmation that it works. 😄

Here's the binary I built while testing with `OLED_PANEL_TINYBOY2` enabled: [Wanhao-D6-Marlin-2.0.x-f646499.hex.zip](https://github.com/MarlinFirmware/Marlin/files/3925303/Wanhao-D6-Marlin-2.0.x-f646499.hex.zip)

**Comment by Deneteus** [comment]
Got it! Just gonna go flash it!

**Comment by Deneteus** [comment]
I am waiting on a netbook that I use to update firmware to finish restarting from a Windows Update. It  had been going for almost 8hrs. LOL

**Comment by Deneteus** [comment]
@thisiskeithb Alright it flashed. LCD works. I just need to build it with the Tiny LCD settings and different settings for a Bondtech extruder. 

![image](https://user-images.githubusercontent.com/6380390/70212461-eeab6100-16fc-11ea-9b10-074f56ebaa19.png)

![image](https://user-images.githubusercontent.com/6380390/70212442-e8b58000-16fc-11ea-9e65-d2ec7f7f83f8.png)

**Comment by thisiskeithb** [comment]
Thanks for finding this bug 🙂

PR #16117 will fix the default D6 config.

Edit: OLED screens sure are nice.

**Comment by Deneteus** [comment]
Do you know if there is an issue related to this? 

![image](https://user-images.githubusercontent.com/6380390/70214850-beb28c80-1701-11ea-9f83-79f79b9f5c29.png)

**Comment by Deneteus** [comment]
I got my printer updated with all of the changes I needed. Rob's config files had some tweaks to the default feed rates, adaptive step smoothing and power loss recovery. 

I enabled S_CURVE_acceleration, Bondtech ESTEPS, Tuned A4988 Stepper Defaults, Raised the max temp since I upgraded to Microswiss All Metal.

[Deneteus Marlin 2.0 D6 Configs.zip](https://github.com/MarlinFirmware/Marlin/files/3925776/Deneteus.Marlin.2.0.D6.Configs.zip)
